zipline/assets/assets.py

```diff
@@ -492,6 +492,74 @@ def lookup_future_chain(self, root_symbol, as_of_date):
 
         return list(map(self._retrieve_futures_contract, sids))
 
+    def lookup_expired_futures(self, start, end):
+        start = start.value
+        end = end.value
+
+        fc_cols = self.futures_contracts.c
+
+        sids = list(map(
+            itemgetter('sid'),
+            sa.select((fc_cols.sid,)).where(
+
+                    # Filter to contracts that are still valid. If both
+                    # exist, use the one that comes first in time (i.e.
+                    # the lower value). If either notice_date or
+                    # expiration_date is NaT, use the other. If both are
+                    # NaT, the contract cannot be included in any chain.
+                    sa.case(
```

Comment:
These two case statements are both trying to find the same date, we could do something more concise with:
```python
nd = sa.func.nullif(fc_cols.notice_date, pd.tslib.iNaT)
ed = sa.func.nullif(fc_cols.expiration_date, pd.tslib.iNaT)
date = sa.func.coalesce(sa.func.min(nd, ed), ed, nd)

whereclause = (date >= start) & (date <= end)
```

We can use this same `nd` and `ed` for the order by:

```
sa.func.coalesce(ed, nd)
```